src/frontend/screens/Sync/index.tsx

```diff
@@ -0,0 +1,211 @@
+import {useWifiName} from '../../hooks/useWifiStatus';
+import {NativeNavigationComponentNoHeader} from '../../sharedTypes';
+import {Text} from '../../sharedComponents/Text';
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
+import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
+import {BLACK, WHITE} from '../../lib/styles';
+import {SettingsButton} from '../ObservationsList/SettingsButton';
+import {Button} from '../../sharedComponents/Button';
+import {useProject, useProjectSettings} from '../../hooks/server/projects';
+import {State} from '@mapeo/core/dist/sync/sync-api';
+import {useDeviceInfo} from '../../hooks/server/deviceInfo';
+import {defineMessages, useIntl} from 'react-intl';
+
+const m = defineMessages({
+  sync: {
+    id: 'screens.SyncModal.SyncView.sync',
+    defaultMessage: 'Sync',
+    description: 'Button to start Synchronization',
+  },
+  stop: {
+    id: 'screens.SyncModal.SyncView.stop',
+    defaultMessage: 'Stop',
+    description: 'Button to stop Synchronization',
+  },
+  caughtUp: {
+    id: 'screens.SyncModal.SyncView.caughtUp',
+    defaultMessage: "You're all caught up!",
+  },
+  yourName: {
+    id: 'screens.SyncModal.SyncView.yourName',
+    defaultMessage: 'Your device name is {deviceName}',
+  },
+  connectedPeers: {
+    id: 'screens.SyncModal.SyncView.connectedPeers',
+    defaultMessage: '{numberOfDevices} Devices nearby/connected',
+  },
+  numberWaitingSync: {
+    id: 'screens.SyncModal.SyncView.numberWaitingSync',
+    defaultMessage: '{numberOfDevices} Devices',
+    description: "Sentence is 'x devices' waiting to sync with you",
+  },
+  waitingToSync: {
+    id: 'screens.SyncModal.SyncView.waitingToSync',
+    defaultMessage: 'Waiting to Sync with you',
+    description: "Sentence is x devices 'waiting to sync with you'",
+  },
+  noDevices: {
+    id: 'screens.SyncModal.SyncView.noDevices',
+    defaultMessage: 'No Device are Syncing',
+  },
+});
+
+/**
+ *
+ * States needed:
+ * - syncing
+ * - not syncing
+ *    - devices waiting to sync
+ *    - data up to date
+ */
+
+export const Sync: NativeNavigationComponentNoHeader<'Sync'> = ({
+  navigation,
+}) => {
+  const ssid = useWifiName();
+  const sync = useProject().$sync;
+  const projectSettings = useProjectSettings();
+  const deviceInfo = useDeviceInfo();
+  const [connectedPeers, setConnectedPeers] = React.useState<number>(0); //this should initialize with `sync.getState().connectPeers`, but it is undefined
+  const [isSyncing, setIsSyncing] = React.useState(false);
+  const [hasDataToSync, setHasDataToSync] = React.useState<boolean>(); //this should initialize with `sync.getState().data.dataToSync`, but it is undefined
+  const {formatMessage} = useIntl();
+
+  console.log(sync.getState()); // sync.getState() === {"_h": 0, "_i": 0, "_j": null, "_k": null, "clear": [Function anonymous]}
+
+  React.useEffect(() => {
+    const syncListener = (val: State) => {
```

Comment:
I don't see the advantage of splitting the state into separate pieces. It's just going to mean as we add features here we keep needing to add additional states, and it introduces the possibility of these different states getting out of sync, which they shouldn't - there is one sync state.